src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -1,107 +1,69 @@
-        private readonly IModel _subscriberChannel;
+        private readonly ConnectionFactory _factory;
+        private IConnection _publisherClient;
+        private IConnection _subscriberClient;
+        private IModel _publisherChannel;
+        private IModel _subscriberChannel;
+        private bool _delayedExchangePluginEnabled = true;
         private readonly TimeSpan _defaultMessageTimeToLive = TimeSpan.MaxValue;
 
         /// <summary>
-        /// Constructor for RabbitMqMessaging - Exchange type set as Direct exchange that uses the routing key
+        /// Constructor for RabbitMqMessaging - Exchange type set as fanout exchange that uses the exchange name
         /// </summary>
         /// <param name="connectionString">The connection string. See https://www.rabbitmq.com/uri-spec.html for more information.</param>
-        /// <param name="queueName">Queue name</param>
+        /// <param name="queueName">Name of the queue established by the subscriber when they call QueueDeclare. Its not used by publisher.</param>
         /// <param name="routingKey">The routing key is an "address" that the exchange may use to decide how to route the message</param>
         /// <param name="exhangeName">Name of the direct exchange that delivers messages to queues based on a message routing key</param>
-        /// <param name="durable">Durable exchanges survive broker restart</param>
-        /// <param name="autoDelete">True, if you want the queue to be deleted when the connection is closed</param>
         /// <param name="queueArguments">queue arguments</param>
         /// <param name="defaultMessageTimeToLive">The value of the expiration field describes the TTL period in milliseconds</param>
         /// <param name="serializer">For data serialization</param>
         /// <param name="loggerFactory">logger</param>
-        /// <param name="persistent">When set to true, RabbitMQ will persist message to disk</param>
-        /// <param name="exclusive"></param>
-        public RabbitMQMessageBus(string connectionString, string queueName, string routingKey, string exhangeName, bool durable = true, bool persistent = true, bool exclusive = false, bool autoDelete = false, IDictionary<string, object> queueArguments = null, TimeSpan? defaultMessageTimeToLive = null, ISerializer serializer = null, ILoggerFactory loggerFactory = null) : base(loggerFactory) {
+        /// <remarks>https://www.rabbitmq.com/dotnet-api-guide.html#connection-recovery</remarks>
+        public RabbitMQMessageBus(string connectionString, string queueName, string routingKey, string exhangeName, IDictionary<string, object> queueArguments = null, TimeSpan? defaultMessageTimeToLive = null, ISerializer serializer = null, ILoggerFactory loggerFactory = null) : base(loggerFactory) {
             _serializer = serializer ?? new JsonNetSerializer();
             _exchangeName = exhangeName;
+            _routingKey = routingKey ?? String.Empty;
```

Comment:
@barkhashah the only change I made from your commit on August 3rd was I noticed you were not setting the routing key via the constructor so I added this in. Is there any reason you didn't do this? All unit tests started passing after I deleted my existing queues and exchanges via management (http://localhost:15672/#/)